### PR TITLE
Adding optionally shown request payload and headers to output

### PR DESCRIPTION
**Pull request by shaunbruno**
Note - using array join to keep the spacing/formatting consistent with consideration to optional parameters

### CODE REVIEW

**Comment by shaunbruno** [comment]
Associated test expect lines may just need to be updated... My quick attempt at installing/running **lab** output some error about not having the correct version of Hapi. I downloaded the latest from npm???

**Comment by arb** [comment]
What is currently in the "master" branch is not what is in npm. You'll need to checkout the v4.1.0 tag before trying to make this change.

That being said, what is the purpose of this PR? If you want to log the request payload and headers, that should be controlled via an option, not the presence or absence of a value. 

Also that is a lot of information to try to cram onto the console. Wouldn't logging to a file make more sense if you want all that information?

**Comment by shaunbruno** [comment]
Hi Adam,

Thanks for the tip WRT the tag - I'll take a look there when I get a chance.

The motivation for this PR has to do with the following:
https://github.com/hapijs/good/issues/299

The issue is the top level (Good plugin) forwards options of logResponsePayload, logRequestPayload, logHeaders (recently renamed to responsePayload, requestPayload, and requestHeaders) to the reporters, but those options are not being utilized in a meaningful way (Good-Console ignores the passed headers and request payload, however uses the response payload). Additionally, it is undocumented as to whether any/all reporters utilize these forwarded (top-level) options, so when all these options are turned on, for example, Good-Console prints the response payload, but silently no-ops for the request and headers.  As the ticket above suggests, this indeed creates confusion - not to mention frustration for new users. 

Currently, as the Good plugin controls these 3 options, and forwards the data (not the options) to the reporters, you'll notice my commit piggybacked on the current practice of logging based on the presence or absence of these values - prior to my commit the response payload is shown if Good forwarded the top level option to Good-Console (and therefore if Good-Console has *that* value, it shows it). 

As the architecture for Good/Good-Console currently stands, the most obvious behavior for top-level options is that each reporter would uniformly respect it (turning on *request payload* at Good plugin level should show *request payload data* in any report that is connected).  Alternatively, and arguably better, these 3 options should be at the reporter level (as an interface the Good plugin expects), where each reporter can subscribe to receiving this data if the options are set for *that specific* reporter. 

**Example (Current)**
```
var options = {
    reporters: [{
        reporter: require('good-console'),
        args: [{ log: ['error', 'medium'] }]
    }],
    logRequestHeaders: true,
    logRequestPayload:true,
    logResponsePayload: true
};
```
**Could Become**
````
var options = {
    reporters: [{
        reporter: require('good-console'),
        args: [
            { log: ['error', 'medium'], 
              logRequestHeaders: true, 
              logRequestPayload: true, 
              logResponsePayload: true
           }]
    }]
};
```
Or something similar (where the options are passed to the reporters directly). However, this change is an architecture shift and beyond the scope of this PR, as it requires buy in and modification to Good Plugin, Good-Console, Good-File, and any other reporter.

Therefore, this PR is only concerned with adhering to the current architecture in an intuitive manner, specifically that top-level options are ubiquitously respected (particularly by Good-Console).

WRT the magnitude of information sent to the console - it's definitely true I would be more apt to send heavy data to a file for production environments, but the additional info is fantastic for development (I see an opportunity for a pretty print option). Closing the loop, if each reporter *did not* support the full set of logging features, as a developer and user of Good and it's reporters, I'll simply choose the most beneficial reporter and redirect it's output (i.e. I'll tail the output from Good-File and get my console, or I'll redirect my Good-Console to a file and get my file). But if users start doing this, that defeats the point of having the reporters altogether, so shouldn't each reporter have (at least) some interpretation of each data point it's aware of? 

Cheers

**Comment by arb** [comment]
@shaunbruno lets go with ability to pass in these options into good-console rather than only relying on the presence or absence of the data.

Clean this PR up and we can incorporate this into the next version. 

[] - add tests
[] - add documentation

**Comment by shaunbruno** [comment]
This PR is complete and ready for pull.

File: README.md [hunk 1]
```diff
@@ -19,6 +19,8 @@ Creates a new GoodConsole object with the following arguments:
 - `[options]` - optional object with the following available keys
 	- `format` - [MomentJS](http://momentjs.com/docs/#/displaying/format/) format string. Defaults to 'YYMMDD/HHmmss.SSS'.
 	- `utc` - boolean controlling Moment using [utc mode](http://momentjs.com/docs/#/parsing/utc/) or not. Defaults to `true`.
+	- `logHeaderPayloadWhenProvided` - boolean controlling whether header data is logged as part of GreatResponse events emitted by Good. Good [header option](https://github.com/hapijs/good#greatresponserequest-options) must also be enabled to forward header data to Good-Console. Defaults to `false`.
```
Comment (arb) [inline README.md]: Too wordy. Just `logHeader` and `logPayload`.

File: lib/index.js [hunk 2]
```diff
@@ -11,7 +11,9 @@ var Through = require('through2');
 var internals = {
     defaults: {
         format: 'YYMMDD/HHmmss.SSS',
-        utc: true
+        utc: true,
+        logHeaderPayloadWhenProvided: false,
```
Comment (arb) [inline lib/index.js]: Same as above.

File: lib/index.js [hunk 3]
```diff
@@ -141,13 +158,30 @@ internals.GoodConsole.prototype._formatResponse = function (event, tags) {
...
+        output.push(statusCode);
+    }
+
+    // response time
+    output.push(Hoek.format('(%sms)', event.responseTime));
+
+    // headers
+    if (this._settings.logHeaderPayloadWhenProvided &&      // option to include header payload enabled
+      typeof event.headers === 'object' && event.headers) { // header payload actually provided by Hapi
+      var headerPayload = 'headers:' + SafeStringify(event.headers);
```
Comment (arb) [inline lib/index.js]: Don't need `SafeStringify` here. Just regular JSON.stringify.

**Comment by arb** [comment]
@shaunbruno please review the style guide [here](https://github.com/hapijs/contrib/blob/master/Style.md). You are using two spaces instead of four throughout.

**Comment by arb** [comment]
Trumped by https://github.com/hapijs/good-console/pull/37
